### PR TITLE
Dns provider fix

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

Removing cross account dns roles in terragrunt.hcl when the environment is staging or production. They are not needed since the DNS zone that is being modified is in the same account. This also fixes the chicken/egg scenario where terragrunt fails to apply because it requires account roles that don't yet exist.

# Test instructions | Instructions pour tester la modification

Terragrunt apply in staging/production work as expected

### CODE REVIEW

File: env/terragrunt.hcl [hunk 1]
```diff
@@ -1,5 +1,7 @@
 locals {
   vars = read_terragrunt_config("../env_vars.hcl")
+  # dns_role is very fragile - if not set exactly as below, terraform fmt will fail in github actions.
+  dns_role           = local.vars.inputs.env == "production" || local.vars.inputs.env == "staging" ? "" : "\n  assume_role {\n    role_arn = \"arn:aws:iam::${local.vars.inputs.dns_account_id}:role/${local.vars.inputs.env}_dns_manager_role\"\n  }"
```
Comment (jimleroyer) [inline env/terragrunt.hcl:4]: 🧙

File: env/terragrunt.hcl [hunk 2]
```diff
@@ -1,5 +1,7 @@
 locals {
   vars = read_terragrunt_config("../env_vars.hcl")
+  # dns_role is very fragile - if not set exactly as below, terraform fmt will fail in github actions.
```
Comment (jimleroyer) [inline env/terragrunt.hcl:3]: Can we also add a comment on why we need this?

File: env/terragrunt.hcl [hunk 3]
```diff
@@ -105,6 +105,11 @@ variable "cbs_satellite_bucket_name" {
   description = "Name of the Cloud Based Sensor S3 satellite bucket"
   type        = string
 }
+variable "cross_account_dns" {
+  description = "Set to true if the DNS zone is not in the same account as the target aws account"
+  type        = bool
+  default     = true
+}
```
[the remark this answers is not in the document]
Comment (ben851) [reply env/terragrunt.hcl:112]: Whoops this is test code. I will remove.

**Review by jimleroyer — APPROVED** [review]
This is quite satisfying as the amount of changes are way less than previous attempts and that is more elegant as well. Good job!

**Pull request merged**